### ISSUE TITLE
🐞 Dagger log deduplication

### ISSUE THREAD

**Issue by grouville** [dagger/dagger#2058]
### What is the issue?

@jpluscplusm reported that he had a log deduplication issue on his Dagger Plan, both locally and on GHA.

After checking it, It's the exact same behavior as #449, but no `#Push` operation is involved in the process 🤔 I personally didn't believe this was possible.

Making it an issue so that we don't lose track of that.

### Log output

```
Downloading https://dl.dagger.io/dagger/releases/0.2.4/dagger_v0.2.4_linux_amd64.tar.gz
Extracting Dagger
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/cf3b7bc8-2a12-485b-b746-66c734b1fd5a -f /home/runner/work/_temp/432d005d-7f15-4e[9](https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true#step:4:9)d-ba0b-590bbf7859d2
/opt/hostedtoolcache/dagger/0.2.4/x64/dagger do test --log-format plain
time="2022-04-02T20:44:05Z" level=warning msg="commandConn.CloseWrite: commandconn: failed to wait: signal: terminated"
time="2022-04-02T20:44:05Z" level=warning msg="commandConn.CloseRead: commandconn: failed to wait: signal: terminated"
time="2022-04-02T20:44:05Z" level=warning msg="commandConn.CloseWrite: commandconn: failed to wait: signal: terminated"
8:44PM INF actions._htmltest | computing
8:44PM INF client.filesystem.".".read | computing
8:44PM INF actions._hugo | computing
8:44PM INF actions._htmltest | completed    duration=800ms
8:44PM INF actions._hugo | completed    duration=800ms
8:44PM INF client.filesystem.".".read | completed    duration=7.7s
8:44PM INF actions.build.run | computing
8:44PM INF actions.build.run | #7 0.[10](https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true#step:4:10)9 Building sites …
8:44PM INF actions.build.run | WARN 2022/04/02 20:44:14 Page's .Hugo is deprecated and will be removed in a future release. Use the global hugo function.
8:44PM INF actions.build.run | #7 0.527 WARN 2022/04/02 20:44:14 Page's .RSSLink is deprecated and will be removed in a future release. Use the Output Format's link, e.g. something like:
8:44PM INF actions.build.run | #7 0.527     {{ with .OutputFormats.Get "RSS" }}{{ .RelPermalink }}{{ end }}.
8:44PM INF actions.build.run | #7 0.529 WARN 2022/04/02 20:44:14 Page's .URL is deprecated and will be removed in a future release. Use .Permalink or .RelPermalink. If what you want is the front matter URL value, use .Params.url.
8:44PM INF actions.build.run | #7 2.357
8:44PM INF actions.build.run | #7 2.357                    |  EN
8:44PM INF actions.build.run | #7 2.358 +------------------+------+
8:44PM INF actions.build.run | #7 2.358   Pages            |  7[11](https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true#step:4:11)
8:44PM INF actions.build.run | #7 2.358   Paginator pages  |   [12](https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true#step:4:12)
8:44PM INF actions.build.run | #7 2.358   Non-page files   |    0
8:44PM INF actions.build.run | #7 2.359   Static files     | 1044
8:44PM INF actions.build.run | #7 2.359   Processed images |    0
8:44PM INF actions.build.run | #7 2.359   Aliases          |  232
8:44PM INF actions.build.run | #7 2.359   Sitemaps         |    1
8:44PM INF actions.build.run | #7 2.360   Cleaned          |    0
8:44PM INF actions.build.run | #7 2.361
8:44PM INF actions.build.run | #7 2.361 Total in 2252 ms
8:44PM INF actions.build.run | completed    duration=5.6s
8:44PM INF actions.build.result._copy | computing
8:44PM INF actions.build.result._copy | completed    duration=1.6s
8:44PM INF actions.test | computing
8:44PM INF client.filesystem.docs.write | computing
8:44PM INF actions.build.run | #7 0.109 Building sites … WARN 2022/04/02 20:44:[14](https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true#step:4:14) Page's .Hugo is deprecated and will be removed in a future release. Use the global hugo function.
8:44PM INF actions.build.run | #7 0.527 WARN 2022/04/02 20:44:14 Page's .RSSLink is deprecated and will be removed in a future release. Use the Output Format's link, e.g. something like:
8:44PM INF actions.build.run | #7 0.527     {{ with .OutputFormats.Get "RSS" }}{{ .RelPermalink }}{{ end }}.
8:44PM INF actions.build.run | #7 0.529 WARN 2022/04/02 20:44:14 Page's .URL is deprecated and will be removed in a future release. Use .Permalink or .RelPermalink. If what you want is the front matter URL value, use .Params.url.
8:44PM INF actions.build.run | #7 2.357
8:44PM INF actions.build.run | #7 2.357                    |  EN
8:44PM INF actions.build.run | #7 2.358 +------------------+------+
8:44PM INF actions.build.run | #7 2.358   Pages            |  711
8:44PM INF actions.build.run | #7 2.358   Paginator pages  |   12
8:44PM INF actions.build.run | #7 2.358   Non-page files   |    0
8:44PM INF actions.build.run | #7 2.359   Static files     | 1044
8:44PM INF actions.build.run | #7 2.359   Processed images |    0
8:44PM INF actions.build.run | #7 2.359   Aliases          |  232
8:44PM INF actions.build.run | #7 2.359   Sitemaps         |    1
8:44PM INF actions.build.run | #7 2.360   Cleaned          |    0
8:44PM INF actions.build.run | #7 2.361
8:44PM INF actions.build.run | #7 2.361 Total in 2252 ms
8:44PM INF client.filesystem.docs.write | completed    duration=5.5s
8:44PM INF actions.test | #9 0.193 htmltest started at 08:44:21 on /test
8:44PM INF actions.test | #9 0.193 ========================================================================
8:44PM INF actions.test | #9 1.922 ✔✔✔ passed in 1.7327465[16](https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true#step:4:16)s
8:44PM INF actions.test | #9 1.923 tested 7[22](https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true#step:4:22) documents
8:44PM INF actions.test | #9 1.9[23](https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true#step:4:23) 
8:[44](https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true#step:4:44)PM INF actions.test | completed    duration=5.6s
```

### Steps to reproduce

Here is the config that reproduces this issue:
```cue
package omnomfrickinnom

import (
	"list"
	"dagger.io/dagger"
	"dagger.io/dagger/core"
)

dagger.#Plan & {
	client: filesystem: {
		".": read: exclude: [ ".git", "docs"]
		docs: write: contents: actions.build.result.output
	}

	actions: {
		_hugo:        core.#Pull & {source: "klakegg/hugo:0.55.1"}
		_hugoImage:   _hugo.output
		build: {
			run: core.#Exec & {
				input: _hugoImage
				mounts: repo: {
					dest:     "/repo/"
					type:     "fs"
					contents: client.filesystem.".".read.contents
				}
				workdir: mounts.repo.dest
				args: [
					"hugo", // "--log", "--debug", "--verboseLog", "--verbose",
					"--destination", result.path,
					"--cleanDestinationDir",
					"--gc",
				]
			}
			result: core.#Subdir & {
				input: run.output
				path:  "/outdir/"
			}
		}
		_htmltest:      core.#Pull & {source: "wjdp/htmltest"}
		_htmltestImage: _htmltest.output
		test:           core.#Exec & {
			input: _htmltestImage
			mounts: repo: {
				dest:     "/repo/"
				type:     "fs"
				contents: client.filesystem.".".read.contents
			}
			mounts: result: {
				dest:     "/test/"
				type:     "fs"
				contents: build.result.output
			}
			workdir: mounts.repo.dest
			args:    list.FlattenN([ _htmltest.config.entrypoint, mounts.result.dest, "-c", "./.htmltest"], 1)
		}
	}
}
```
src: https://github.com/jpluscplusm/omnomfrickinnom/blob/242e7a2d21d16a92c1dbe9021f190df122071ebf/dagger.cue
CI: https://github.com/jpluscplusm/omnomfrickinnom/runs/5801819483?check_suite_focus=true

### Dagger version

dagger 0.2.4

### OS version

Irrelevant

**Comment by jpluscplusm** [comment]
Apologies for not filing this Issue myself - I've been working on paring the repro down, as the omnom repo linked belongs to a friend, and contains several hundred MB of binary assets. The build process, through the lens of my very imperfect Dagger knowledge, is bulky and slow, and I didn't want to highlight this problem until I'd both figured out that it /was/ a problem, and how to make it reproducible less painfully by other folks. 

I'll comment here when I manage to extract that minimal repro case.

**Comment by sipsma** [comment]
Just a note that this is likely fixed by #2267. The bug that duplicates is caused by any export and in this case there appears to be an export to a local filesystem happening.

**Comment by sipsma** [comment]
Resolving as #2267 has been merged, it has a test case using local exports that previously resulted in duplicated logs and now works as expected
